geniza/corpus/management/commands/googledoc_to_annotation.py

```diff
@@ -0,0 +1,389 @@
+            )
+            for doc in self.normalized_unicode:
+                print("\t%s" % doc)
+
+        if self.document_not_found:
+            print(
+                "Document not found for %s PGPID%s:"
+                % (len(self.document_not_found), pluralize(self.document_not_found))
+            )
+            for doc in self.document_not_found:
+                print("\t%s" % doc)
+
+    def get_credentials(self):
+        SCOPES = ["https://www.googleapis.com/auth/drive.readonly"]
+        creds = None
+        # The token json file stores the user's access and refresh tokens, and is
+        # created automatically when the authorization flow completes for the first
+        # time.
+        if os.path.exists(settings.GOOGLE_API_TOKEN_FILE):
+            creds = Credentials.from_authorized_user_file(
+                settings.GOOGLE_API_TOKEN_FILE, SCOPES
+            )
+        # If there are no (valid) credentials available, let the user log in.
+        if not creds or not creds.valid:
+            if creds and creds.expired and creds.refresh_token:
+                creds.refresh(Request())
+            else:
+                flow = InstalledAppFlow.from_client_secrets_file(
+                    settings.GOOGLE_API_SECRETS_FILE, SCOPES
+                )
+                creds = flow.run_local_server(port=0)
+            # Save the credentials for the next run
+            with open(settings.GOOGLE_API_TOKEN_FILE, "w") as token:
+                token.write(creds.to_json())
+        return creds
+
+    def download_as_html(self, file_id):
+        # export file to HTML
+        request = self.service.files().export_media(
+            fileId=file_id, mimeType="text/html"
+        )
+        file = io.BytesIO()
+
+        # start the download, then loop through all chunks until file is downloaded
+        downloader = MediaIoBaseDownload(file, request)
+        done = False
+        while done is False:
+            _, done = downloader.next_chunk()
+
+        return file.getvalue()
+
+    def new_translation_annotation(self):
+        # initialize a new annotation dict object with all the defaults set
+
+        anno = Dict()
+        setattr(anno, "@context", "http://www.w3.org/ns/anno.jsonld")
+        anno.type = "Annotation"
+        anno.body = [Dict()]
+        anno.body[0].type = "TextualBody"
+        anno.body[0].format = "text/html"
+        # supplement rather than painting over the image
+        # add translating as secondary motivation
+        anno.motivation = ["sc:supplementing", "translating"]
+
+        anno.target.source.type = "Canvas"
+        anno.target.selector.type = "FragmentSelector"
+        anno.target.selector.conformsTo = "http://www.w3.org/TR/media-frags/"
+
+        return anno
+
+    def process_file(self, name, html_file):
+        soup = BeautifulSoup(html_file, "html.parser")
+        # first table is metadata, second table is translation
```

Comment:
can you add some validation / check that the file has the supported structure? what happens if someone accidentally drops a file into this folder that doesn't belong there?